### PR TITLE
Improve reusability for RCTRootViewFactory

### PR DESCRIPTION
**Pull request by Kudo**
## Summary:

RCTRootViewFactory is a great work for creating react binding view. we want to reuse the factory inside expo and would be good to have these improvements.
- exposing `reactHost` property so that we can update the RCTHost instance without recreate a factory.
- break bridgeless creation logic to a specific `createReactHost`, so that we can reuse the method for RCTHost creation
 
## Changelog:

[IOS][CHANGED] - Improve reusability for RCTRootViewFactory

## Test Plan:

this pr should not introduce any regression and getting all ci passed

### CODE REVIEW

**Comment by Kudo** [comment]
the three ci failures seems happening on main as well.

**Comment by Kudo** [comment]
cc @okwasniewski if that makes sense to you.

File: packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.h
Comment (okwasniewski) [inline packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.h:101]: I'm good with this change, looks good - let's wait for @cipolleschi approval.

Comment (cipolleschi) [reply packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.h:101]: It looks good to me too, but to reuse the `createReactHost` method, shouldn't it be exposed in the header as well, with:
```objc
- (RCTHost *)createReactHost;
```
?

**Review by cipolleschi — APPROVED** [review]
Good to me, but I think it is missing exposing the `createReactHost` method if you want to reuse it.

**Comment by Kudo** [comment]
> Good to me, but I think it is missing exposing the `createReactHost` method if you want to reuse it.

i was thinking using a category to do that. but that sounds better for maintenance. thanks i've updated the diff.

**Review by cipolleschi — APPROVED** [review]

**Pull request closed without merging**
